Opencv failed to build due to error C2512 on windows with MSVC

Environment:
Windows Server 2016 + VS2017 opencv master branch latest srouce code

Opencv failed to build due to error C2512: 'cv::gapi::GNetPackage': no appropriate default constructor available on windows with MSVC.  It can be first reproduced on 0757a51 reversion on master branch. Could you please look at this issue? Thanks a lot!

Steps to reproduce the behavior:
1.git clone https://github.com/opencv/opencv D:\OpenCV\src
2.git clone https://github.com/opencv/opencv_extra D:\OpenCV\src\extra
3.Open a VS 2017 x86 prompt and browse to D:\OpenCV
4.mkdir build_x86 && pushd build_x86
5.cmake -G "Visual Studio 15 2017" -DCMAKE_SYSTEM_VERSION=10.0.17134.0 -DWITH_IPP=ON -DBUILD_SHARED_LIBS=OFF -DBUILD_PERF_TESTS=ON -DBUILD_TESTS=ON -DBUILD_EXAMPLES=ON -DWITH_OPENCL=OFF -DBUILD_DOCS=OFF -DWITH_CUDA=OFF ..\src
6.msbuild /p:Configuration=Release;Platform=Win32 build_x86\OpenCV.sln /t:Rebuild /m /p:BuildInParallel=true

The whole log file please see attachment.
[log_x86_build.log](https://github.com/opencv/opencv/files/3475024/log_x86_build.log)

ErrorMessage:
D:\OpenCV\src\modules\gapi\src\compiler\gcompiler.cpp(115): error C2512: 'cv::gapi::GNetPackage': no appropriate default constructor available [D:\OpenCV\build_x86\modules\gapi\opencv_gapi.vcxproj]

--- a/modules/gapi/include/opencv2/gapi/infer.hpp
+++ b/modules/gapi/include/opencv2/gapi/infer.hpp
@@ -199,7 +199,8 @@ struct GAPI_EXPORTS GNetParam {
  * @sa cv::gapi::networks
  */
 struct GAPI_EXPORTS GNetPackage {
-    explicit GNetPackage(std::initializer_list<GNetParam> &&ii = {});
+    GNetPackage() : GNetPackage({}) {}
+    explicit GNetPackage(std::initializer_list<GNetParam> &&ii);
     std::vector<GBackend> backends() const;
     std::vector<GNetParam> networks;
 };

